src/agentconfig.rs

```diff
@@ -228,6 +289,51 @@ impl Configurable for AgentConfig {
         };
         info!("Selected {} as local address to listen on.", final_ip);
 
+        // Parse data directory from values
+        let final_data_dir = if let Ok(data_dir) =
```

Comment:
If we add one more it makes sense to move it into a function. This is a 1:1 copy of the stuff below

Reply:
I considered it, but opted for "doesn't have to look nice at this stage" - which incidentally you keep telling me ;)

Reply:
I pulled the code into a fn and refactored it a bit.